src/index.js

```diff
@@ -12,93 +12,99 @@ module.exports = function (serverless) {
+                                    Version: '2012-10-17',
+                                    Statement: [
+                                        {
+                                            Effect: 'Allow',
+                                            Action: [
+                                                'logs:CreateLogGroup',
+                                                'logs:CreateLogStream',
+                                                'logs:DescribeLogGroups',
+                                                'logs:DescribeLogStreams',
+                                                'logs:PutLogEvents',
+                                                'logs:GetLogEvents',
+                                                'logs:FilterLogEvents'
+                                            ],
+                                            Resource: '*'
+                                        }
+                                    ]
+                                }
                             }
-                        }
-                    },
-                    ApiGatewayAccount: {
-                        Type: 'AWS::ApiGateway::Account',
-                        Properties: {
-                            CloudWatchRoleArn: {
-                                'Fn::GetAtt': [
-                                    logRoleLogicalName,
-                                    'Arn'
+                        ],
+                        Path: '/',
+                        RoleName: {
+                            'Fn::Join': [
+                                '-',
+                                [
+                                    serverless.service.service,
+                                    serverless.getProvider('aws').getStage(),
+                                    serverless.getProvider('aws').getRegion(),
+                                    'apiGatewayLogRole'
                                 ]
-                            }
-                        },
-                        DependsOn: [
-                            logRoleLogicalName
-                        ]
+                            ]
+                        }
                     }
                 },
-
+                ApiGatewayAccount: {
+                    Type: 'AWS::ApiGateway::Account',
+                    Properties: {
+                        CloudWatchRoleArn: {
+                            'Fn::GetAtt': [
+                                logRoleLogicalName,
+                                'Arn'
+                            ]
+                        }
+                    },
+                    DependsOn: [
+                        logRoleLogicalName
+                    ]
+                }
+            };
+            const hasCloudFormationSettings = methodSettings.some(setting =>
+                setting === undefined ||
+                ['DataTraceEnabled', 'MetricsEnabled', 'LoggingLevel'].some(prop => setting[prop])
```

Comment:
A couple of specific points here:

* `LoggingLevel` can have a value of `OFF`; if the logging permissions are only being applied when logging is enabled, then it seems reasonable that they wouldn't be applied if the value is `OFF` (maybe?)
* As per my overall comment -- and this isn't something you've introduced but rather uncovered -- this should actually be split into two groups, `DataTraceEnabled` and `LoggingLevel` should enable the `logs:*` permissions and `MetricsEnabled` should enable the `cloudwatch:PutMetricData` permission

Reply:
@leftclickben I am still a bit green on some of this permissions stuff but what you're saying so far makes sense. Any new ideas recently?

Reply:
@leftclickben I also realized that what I want to accomplish is possible with serverless-plugin-stage-variables without needing the extra permissions (I was confused at first about how to implement this) so I am sticking with that as a solution to my specific issue. However, if you do want to be more refined about your permissions on this project then hopefully the input helped!